src/components/CalendarButtonTwoStep/CalendarButtonTwoStep.js

```diff
@@ -0,0 +1,163 @@
+import React, { Component } from 'react';
+import { findDOMNode } from 'react-dom';
+import PropTypes from 'prop-types';
+import styled, { css } from 'react-emotion';
+import { START_DATE, END_DATE } from 'react-dates/constants';
+
+import { RangePickerController } from '../Calendar';
+import Button from '../Button';
+import { textKilo } from '../../styles/style-helpers';
+
+class CalendarButton extends Component {
+  state = { startDate: null, endDate: null, focusedInput: null };
+  buttonRef = null;
+
+  _onDatesChange = ({ startDate, endDate }) => {
+    this.setState({ startDate, endDate });
+
+    if (!endDate) {
```

Comment:
We tend to omit the `else` block and instead early return in the `if` block.

Reply:
I've switched it to ternary expression. The if block was because initially I've expected this to have a lot of cases.